### ISSUE TITLE
Add DSLContext.map(Schema) and map(Table) to expose the schema mapping feature

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#5987]
Users currently need to wrap a table in a query to be able to trigger the [runtime schema mapping feature](https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/runtime-schema-mapping/). Sometimes, they prefer to simply get a hold of a mapped schema or table reference (or other objects). For instance:

```java
Table<?> unmapped = ...
Table<?> mapped = configuration.map(table); // This should be possible
assertEquals("some_mapped_name", mapped.getName());
```

It is possible to work around this in various ways, e.g.

- By using the internal `Tools.getMappedTable()` method
- By generating a SQL string and parsing it again
- By calling the deprecated `configuration.schemaMapping().map(table)`

Much better: We'll offer this feature in the public API.

----
See also:
- https://groups.google.com/forum/#!topic/jooq-user/ryr-VoiztEc
- https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/runtime-schema-mapping/

**Comment by cconroy** [comment]
Thanks for filing this!

Note: in addition to getting the table name, it's also important to be able to fetch the schema name.

**Comment by lukaseder** [comment]
[Sure: `Table.getSchema()`](https://www.jooq.org/javadoc/latest/org/jooq/Table.html#getSchema--)

**Comment by lukaseder** [comment]
Egh... We already have this :) Implemented in version 3.0. Closing this as duplicate of https://github.com/jOOQ/jOOQ/issues/2203
